### ISSUE TITLE
escape instead of removing

### ISSUE THREAD

**Issue by iliakan** [punkave/sanitize-html#57]
I remember there was a way to escape instead of removing, but can't find in the docs.

Like: I have a `text <with> <tags>`, and I want it to become `text &lt;with&gt; &lt;tags&gt;`, all tags in-place, but escaped.

**Comment by boutell** [comment]
I don't recall such a feature.

On Tue, Jun 30, 2015 at 7:52 AM, Ilya Kantor <notifications@github.com>
wrote:

> I remember there was a way to escape instead of removing, but can't find
> in the docs.
>
> Like: I have a text <with> <tags>, and I want it to become text
> &lt;with&gt; &lt;tags&gt;, all tags in-place, but escaped.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/punkave/sanitize-html/issues/57>.
>



-- 


*THOMAS BOUTELL, *DEV & OPS
P'UNK AVENUE | (215) 755-1330  |  punkave.com

**Comment by iliakan** [comment]
So, there's no way to kraft it from existing stuff?

Looks like a fairly basic thing.

**Comment by MauriceZ** [comment]
Will this be added any time soon?

**Comment by boutell** [comment]
What is the use case where you want some tags to be respected, but others
to be escaped?


On Sun, Sep 6, 2015 at 12:23 PM, Maurice Zhang <notifications@github.com>
wrote:

> Will this be added any time soon?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/punkave/sanitize-html/issues/57#issuecomment-138097590>
> .
>



-- 


*THOMAS BOUTELL, *DEV & OPS
P'UNK AVENUE | (215) 755-1330  |  punkave.com

**Comment by MauriceZ** [comment]
For example, if a user inputs a script tag, I would want it to appear rather than just deleting it to avoid confusion.

**Comment by iliakan** [comment]
For example, I allow in comments `<a>` and `<i>` markup, but want other tags to appear "as is".

**Comment by boutell** [comment]
It's a pull request I would consider. It doesn't make sense when sanitizing
HTML from a rich text editor etc., where people paste markup from word
processors that it would be better to just discard, but it could make sense
in a context where people are hand-writing markup I suppose.

On Sun, Sep 6, 2015 at 12:52 PM, Ilya Kantor <notifications@github.com>
wrote:

> For example, I allow in comments <a> and <i> markup, but want other tags
> to appear "as is".
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/punkave/sanitize-html/issues/57#issuecomment-138099659>
> .
>



-- 


*THOMAS BOUTELL, *DEV & OPS
P'UNK AVENUE | (215) 755-1330  |  punkave.com

**Comment by dgrad** [comment]
I don't think it's really possible to do this. Let's say you want to allow `<b>` but not allow `<div>`, with this escape instead of remove option turned on. Then something like `Use <b><div></b> to open the DIV tag and <b></div></b> to close it` would be expected to convert to `Use <b>&lt;div&gt;</b> to open the DIV tag and <b>&lt;/div&gt;</b> to close it`. But a string like that would throw off the html parser (and it's hard to follow by a human reader too).

I really think that in a context where people are hand writing markup, interpreting some HTML as HTML and other HTML as plain text would be confusing. You want it to be easy to tell what is markup and what is not. This sounds more like a case where you should be using bbcode or markdown.

**Comment by boutell** [comment]
Yes, the more I think about it you're right. As a user I'm not going to be
mollified by the knowledge that my tags have been escaped rather than
honored (: I don't think it's in scope for this modul.

On Fri, Nov 27, 2015 at 1:10 PM, Daniel Grad <notifications@github.com>
wrote:

> I don't think it's really possible to do this. Let's say you want to allow
> <b> but not allow <div>, with this escape instead of remove option turned
> on. Then something like Use <b><div></b> to open the DIV tag and
> <b></div></b> to close it would be expected to convert to Use
> <b>&lt;div&gt;</b> to open the DIV tag and <b>&lt;/div&gt;</b> to close it.
> But a string like that would throw off the html parser (and it's hard to
> follow by a human reader too).
>
> I really think that in a context where people are hand writing markup,
> interpreting some HTML as HTML and other HTML as plain text would be
> confusing. You want it to be easy to tell what is markup and what is not.
> This sounds more like a case where you should be using bbcode or markdown.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/punkave/sanitize-html/issues/57#issuecomment-160184705>
> .
>



-- 


*THOMAS BOUTELL, *DEV & OPS
P'UNK AVENUE | (215) 755-1330  |  punkave.com

**Comment by iliakan** [comment]
Most of time escaping a tag is more obvious than removing it. Even if strange behavior arises, it's easy to explain what's going on, and that you "should not use tag <script>".

**Comment by boutell** [comment]
I would argue strongly with "most of the time." Manual entry of HTML tags
is an edge case. Using a rich text editor and hitting "paste" from Word is
the common case.

On Tue, Dec 1, 2015 at 1:28 PM, Ilya Kantor <notifications@github.com>
wrote:

> Most of time escaping a tag is more obvious than removing it. Even if
> strange behavior arises, it's easy to explain what's going on, and that you
> "should not use tag
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/punkave/sanitize-html/issues/57#issuecomment-161055320>
> .
>



-- 


*THOMAS BOUTELL, *DEV & OPS
P'UNK AVENUE | (215) 755-1330  |  punkave.com

**Comment by boutell** [comment]
That being said, I suppose optional hooks to do something different with
the discarded tags would not be an unreasonable pull request.

On Tue, Dec 1, 2015 at 2:11 PM, Tom Boutell <tom@punkave.com> wrote:

> I would argue strongly with "most of the time." Manual entry of HTML tags
> is an edge case. Using a rich text editor and hitting "paste" from Word is
> the common case.
>
> On Tue, Dec 1, 2015 at 1:28 PM, Ilya Kantor <notifications@github.com>
> wrote:
>
>> Most of time escaping a tag is more obvious than removing it. Even if
>> strange behavior arises, it's easy to explain what's going on, and that you
>> "should not use tag
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/punkave/sanitize-html/issues/57#issuecomment-161055320>
>> .
>>
>
>
>
> --
>
>
> *THOMAS BOUTELL, *DEV & OPS
> P'UNK AVENUE | (215) 755-1330  |  punkave.com
>



-- 


*THOMAS BOUTELL, *DEV & OPS
P'UNK AVENUE | (215) 755-1330  |  punkave.com
